clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1098,6 +1099,23 @@ public static void writeFully(FileChannel channel, ByteBuffer sourceBuffer) thro
             channel.write(sourceBuffer);
     }
 
+    public static long tryWriteTo(TransferableChannel destChannel,
+                                  long position,
+                                  int length,
+                                  ByteBuffer sourceBuffer) throws IOException {
+        if (position > Integer.MAX_VALUE)
```

Comment:
Is this a restriction of this method itself or are we just carrying over the limitation from `FileRecords`?